gnucash/gnome/gnc-plugin-page-register.c

```diff
@@ -4563,13 +4563,15 @@ gnc_plugin_page_register_cmd_stock_split (GtkAction* action,
                                           GncPluginPageRegister* page)
 {
     Account* account;
+    GtkWindow* window;
 
     ENTER ("(action %p, plugin_page %p)", action, page);
 
     g_return_if_fail (GNC_IS_PLUGIN_PAGE_REGISTER (page));
 
     account = gnc_plugin_page_register_get_account (page);
-    gnc_stock_split_dialog (NULL, account);
+    window = gnc_window_get_gtk_window (GNC_WINDOW (GNC_PLUGIN_PAGE (page)->window));
+    gnc_stock_split_dialog (GTK_WIDGET (window), account);
```

Comment:
@Bob-IT would be good to know if there are possibly any negative consequences of setting parent type properly.

Reply:
There are negative consequences of *not* setting the parent: Instead of the dialog appearing centered in the parent it appears at the root coordinates, usually the top-left corner of the display. On my two-monitor setup that can be quite far away from where I'm looking. If the parent happens to be occupying that space then switching away from GnuCash and back can paint the parent over the dialog, and if the dialog is modal that will make the user think that GnuCash has hung.

Reply:
Ok. I was thinking more in the lines of having a window without a parent, activating the register, modifying data etc. I don't understand the `g_signal` mechanism and wondered if these codes would run inappropriately. I'll push this branch through after release.